relentless/potential/potential.py

```diff
@@ -34,28 +123,52 @@ def evaluate(self, pair):
                 params[p] = v
             elif v is not None:
                 raise TypeError('Parameter type unrecognized')
-
             if v is None:
                 raise ValueError('Parameter {} is not set for ({},{}).'.format(p,pair[0],pair[1]))
 
         return params
 
     def save(self, filename):
+        """Saves the data to a file.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file to save data in.
+
+        """
         all_params = {}
         for key in self:
             all_params[str(key)] = self.evaluate(key)
         with open(filename, 'w') as f:
             json.dump(all_params, f, sort_keys=True, indent=4)
 
     def load(self, filename):
-        raise NotImplementedError('Load is not implemented')
+        """Loads the data from a file.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file from which to load data.
+
+        """
+        with open(filename, 'r') as f:
+            data = json.load(f)
+        for key in data:
+            key_tup = tuple(str(key).strip('"'))
+            if key_tup in self:
+                self[key_tup] = data[key]
```

Comment:
This is heading in the right direction! A few things that will make this more robust:
1. Can you check that the keys of data are _exactly_ the `pairs` of this matrix? That is, they include no more and no less. Otherwise, the JSON schema is invalid.
2. Can you check that the params in each `data[pair]` are exactly those that are in this object? You do check this indirectly with `__setitem__`, but the error message might be confusing to a user calling `load`. Instead, you should raise an exception here that alerts the user that the parameters loaded are inconsistent with the schema of this object.
3. When the parameters are dumped by save, they are evaluated first. This means that we lose information about whether they were stored in a float or a `Variable` inside this class. When you load, you need to detect whether you are working with a scalar or a `Variable`, and then save appropriately. To address this, we probably need to modify the `save` method, so let's discuss tomorrow.
4. Does that combination of tuple str and strip actually explode the key into two strings? I didn't know that! If not, you can go back to using that other tool to process the key.